types/jest/index.d.ts

```diff
@@ -131,6 +131,26 @@ declare namespace jest {
      * Returns the number of fake timers still left to run.
      */
     function getTimerCount(): number;
+    /**
+     * Set the current system time used by fake timers. Simulates a user
+     * changing the system clock while your program is running. It affects the
+     * current time but it does not in itself cause e.g. timers to fire; they
+     * will fire exactly as they would have done without the call to
+     * jest.setSystemTime().
+     *
+     * > Note: This function is only available when using modern fake timers
+     * > implementation
+     */
+    function setSystemTime(now?: number | Date): void;
```

Comment:
As mentioned in the PR description, jest currently types this as:

```ts
    function setSystemTime(now?: number): void;
```

I have [a PR open](https://github.com/facebook/jest/pull/10169) to expand it to support `Date`, given that under the hood `setSystemTime` is calling a function that accepts both. If they do not want to support that signature, I will remove it here.

Reply:
Perhaps we can hold off on adding support for `Date` until the other PR is merged to keep things consistent and work as intended? Once that other PR is merged and a newer version of Jest is released with the `Date` support, we can then update the types here to reflect that latest changes?

Reply:
Sounds good to me